### DISCUSSION TITLE
Some cave missions are missing the "cave" sonar marker

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#13917] [Bug reports]
### Disclaimers 
 
- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Normally when some mission is located in the cave you can see two markers on sonar: mission target and cave
I found 2 MineralMissions with PositionType.Cave that for some reason [weren't added to their cave.MissionsToDisplayOnSonar](https://github.com/FakeFishGames/Barotrauma/blob/master/Barotrauma/BarotraumaShared/SharedSource/Events/Missions/MineralMission.cs#L138)

### Reproduction steps

Check these saves
[no cave sonar marker.zip](https://github.com/FakeFishGames/Barotrauma/files/15187728/no.cave.sonar.marker.zip)

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Single player

### Version

v1.4.5.0 (Blood in the Water Update, hotfix 1)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
~Tested in v1.10.7.2, seems fixed, closing~

**Comment by SomeRandomNoobKekeke** [comment]
No, still borked
Found new bugged save, old ones are not bugged anymore
[Where is cave marker.zip](https://github.com/user-attachments/files/23692970/Where.is.cave.marker.zip)

**Reply by A2ur3** [reply]
Is there a way to manually hide the cave sonar marker?

**Reply by SomeRandomNoobKekeke** [reply]
only with this mod https://steamcommunity.com/sharedfiles/filedetails/?id=3198868074

**Comment by SomeRandomNoobKekeke** [comment]
Ok, nevermind, it's probably just because minerals are spawned outside the cave area rect
Yellow is Cave.Area
![203952~1](https://github.com/user-attachments/assets/0e6ffde1-3cf3-4e8c-9fe1-429c3d226b35)
![20705D~1](https://github.com/user-attachments/assets/48b1b8e8-8000-4093-bb38-2714fb0a4e2d)

**Comment by SoobGorl** [comment]
Ticketed in [#16916](https://github.com/FakeFishGames/Barotrauma/issues/16916)

**Comment by SomeRandomNoobKekeke** [comment]
Another case from dlc now, not sure if the cause is the same
[No Cave Mark.zip](https://github.com/user-attachments/files/27226399/No.Cave.Mark.zip)
